packages/gatsby-plugin-offline/src/gatsby-node.js

```diff
@@ -88,7 +88,7 @@ exports.onPostBuild = (args, pluginOptions) => {
     // URLs and not any files hosted on the site.
     //
     // Regex based on http://stackoverflow.com/a/18017805
-    navigateFallbackWhitelist: [/^[^?]*([^.?]{5}|\.html)(\?.*)?$/],
+    navigateFallbackWhitelist: [/^([^.?]*|[^?]*\.([^.?]{5,}|html))(\?.*)?$/],
```

Comment:
This is getting hard to follow 😱 (not your fault!)

So _originally_ this was for capturing a file name without an extension, so if we break this apart

- start of string
- PARTS
  - `[^.?]*` matches _anything_ without a dot, e.g. `/asdf`
  - `[^?]*\.([^.?]{5,}|html)` matches also html files?
  - `(\?.*)` matches query params
- end of string

Did I encapsulate that OK?